docs/features/tuples.md

```diff
@@ -0,0 +1,227 @@
+
+A lot of details and motivation for the feature is given in 
+[https://github.com/dotnet/roslyn/issues/347](https://github.com/dotnet/roslyn/issues/347)
+
+Current design notes:
+https://github.com/dotnet/roslyn/issues/10429
+
+NOTE: The goal of this document is capture what is being implemented. As design evolves, the document will undergo adjustments. 
+Changes in design will be applied to this document as the changes are implemented.
+
+-----------------------------------
+
+Tuple types
+-----------
+
+Tuple types would be introduced with syntax very similar to a parameter list:
+
+```C#
+public (int sum, int count) Tally(IEnumerable<int> values) { ... }
+    
+var t = Tally(myValues);
+Console.WriteLine($"Sum: {t.sum}, count: {t.count}");   
+ ```
+ 
+The syntax `(int sum, int count)` indicates an anonymous struct type with public fields of the given names and types.
```

Comment:
Probably should say "anonymous data structure with public fields of the given names and types, also referred to as tuple"